include/nuttx/compiler.h

```diff
@@ -381,6 +381,12 @@
 
 #  define UNUSED(a) ((void)(1 || (a)))
 
+#  if defined(__clang__)
+#    define no_builtin(n) __attribute__((no_builtin(n)))
+#  else
+#    define no_builtin(n) __attribute__((__optimize__("-fno-tree-loop-distribute-patterns")))
```

Comment:
I also read that `-ffreestanding` is needed additionally to `-fno-tree-loop-distribute-patterns`, but do not have strong proof of that